python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/router_bgp.schema.yml

```diff
@@ -1439,6 +1439,382 @@ keys:
+                          description: Include community-list references in missing policy decision.
+                        include_prefix_list:
+                          type: bool
+                          description: Include prefix-list references in missing policy decision.
+                        include_sub_route_map:
+                          type: bool
+                          description: Include sub-route-map references in missing policy decision.
+                multi_path:
+                  type: bool
+                next_hop_resolution:
+                  type: dict
+                  keys:
+                    v4_mapped_v6_translation:
+                      type: bool
+                next_hop_self:
+                  type: bool
+                next_hop_self_source_interface:
+                  description: Source interface name.
+                  type: str
+                next_hop_self_v4_mapped_v6_source_interface:
+                  description: v4-mapped-v6 source interface name. Takes precedence over the next_hop_self_source_interface.
+                  type: str
+                next_hop_unchanged:
+                  type: bool
+                rcf_in:
+                  type: str
+                  description: |-
+                    Inbound RCF function name with parenthesis.
+                    Example: MyFunction(myarg).
+                rcf_out:
+                  type: str
+                  description: |-
+                    Outbound RCF function name with parenthesis.
+                    Example: MyFunction(myarg).
+                route_map_in:
+                  type: str
+                  description: Inbound route-map name.
+                route_map_out:
+                  type: str
+                  description: Outbound route-map name.
+          networks:
+            type: list
+            primary_key: prefix
+            min_length: 1
+            items:
+              type: dict
+              keys:
+                prefix:
+                  description: IPv4 prefix "A.B.C.D/E" or IPv6 prefix "A:B:C:D:E:F:G:H/I".
+                  type: str
+                route_map:
+                  description: Route-map name.
+                  type: str
+          next_hops:
+            type: list
+            primary_key: ip_address
+            min_length: 1
+            items:
+              type: dict
+              keys:
+                ip_address:
+                  type: str
+                lfib_backup_ip_forwarding:
+                  type: bool
+          next_hop_resolution_ribs:
+            type: list
+            min_length: 1
+            max_length: 3
+            description: "Specify the RIBs used to resolve next-hops. The order of this list determines the order of RIB lookups."
+            items:
+              type: dict
+              keys:
+                rib_type:
+                  type: str
+                  required: true
+                  valid_values:
```

Comment:
@ClausHolbechArista @gmuloc 
I am in doubt that should we use valid values as list for next-hop resolution ribs because we cant have tunnel-rib multiple times.
```
s2-leaf3(config-router-bgp-af-label)#next-hop resolution rib tunnel-rib asd ?
  system-connected  Resolve using system connected RIB
  tunnel-rib        Resolve using a named tunnel RIB
  <cr>              

s2-leaf3(config-router-bgp-af-label)#next-hop resolution rib tunnel-rib asd tunnel-rib colored ?
  system-colored-tunnel-rib  The system colored tunnel RIB

s2-leaf3(config-router-bgp-af-label)#next-hop resolution rib tunnel-rib asd tunnel-rib ?
  colored  Resolve using a named colored tunnel RIB

s2-leaf3(config-router-bgp-af-label)#next-hop resolution rib tunnel-rib asd tunnel-rib colored system-colored-tunnel-rib ?
  system-connected  Resolve using system connected RIB
  tunnel-rib        Resolve using a named tunnel RIB
  <cr>              

s2-leaf3(config-router-bgp-af-label)#next-hop resolution rib tunnel-rib asd tunnel-rib colored system-colored-tunnel-rib tunnel-rib ?
% Unrecognized command
s2-leaf3(config-router-bgp-af-label)#next-hop resolution rib tunnel-rib asd tunnel-rib colored system-colored-tunnel-rib system-connected ?
  tunnel-rib  Resolve using a named tunnel RIB
  <cr>        

s2-leaf3(config-router-bgp-af-label)#next-hop resolution rib tunnel-rib asd tunnel-rib colored system-colored-tunnel-rib system-connected tunnel-rib ?
% Unrecognized command
s2-leaf3(config-router-bgp-af-label)#next-hop resolution rib tunnel-rib asd tunnel-rib colored system-colored-tunnel-rib system-connected tunnel-rib 
```
I know this will require to change in other AFI and might need to update model. But just an observation

Reply:
Since the order is the key for this feature, a list is still what we want. We could use something like unique keys to ensure an entry is only there once, but let's not go too overboard on this.